src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java

```diff
@@ -23,41 +23,44 @@
 public final class KubernetesProvisioningLimits {
     private static final Logger LOGGER = Logger.getLogger(KubernetesProvisioningLimits.class.getName());
 
-    @GuardedBy("this")
-    private boolean init;
+    private final AtomicBoolean init = new AtomicBoolean();
 
     /**
      * Tracks current number of kubernetes agents per pod template
      */
-    private final Map<String, Integer> podTemplateCounts = new HashMap<>();
+    private final ConcurrentMap<String, Integer> podTemplateCounts = new ConcurrentHashMap<>();
 
     /**
      * Tracks current number of kubernetes agents per kubernetes cloud
      */
-    private final Map<String, Integer> cloudCounts = new HashMap<>();
+    private final ConcurrentMap<String, Integer> cloudCounts = new ConcurrentHashMap<>();
```

Comment:
As the final implementation is keeping the `synchronized` in `this`, I think there is no need to use `ConcurrentMap`s here, but it does not harm either I guess.